mixins/labelled-mixin.js

```diff
@@ -73,16 +91,30 @@ export const LabelledMixin = superclass => class extends superclass {
 		};
 	}
 
-	updated(changedProperties) {
+	constructor() {
+		super();
+		this._missingLabelErrorHasBeenThrown = false;
+		this._throwNoLabelExceptionImmediately = false;
+	}
+
+	async updated(changedProperties) {
 		super.updated(changedProperties);
 
+		if (changedProperties.has('label')) {
+			this._validateLabel();
+		}
+
 		if (!changedProperties.has('labelledBy')) return;
 
 		if (this._labelObserver) this._labelObserver.disconnect();
 
 		if (!this.labelledBy) return;
 
+		// try to get the labelling element syncronously first
 		let labelElem = this.getRootNode().querySelector(`#${cssEscape(this.labelledBy)}`);
+		if (!labelElem) {
+			labelElem = await waitForElement(this.getRootNode(), `#${cssEscape(this.labelledBy)}`, 3000);
```

Comment:
Is there anything in here that can get clobbered if a later lifecycle event is processed? Lit doesn't seem to await its lifecycle events, so if we're waiting 3 seconds here it seems likely that later lifecycle events (incl. re-renders) might start running before this resolves.

Reply:
I have no idea if this is something that might cause a problem in the mixin - it was just my first thought when I saw the long wait. This bit me in the HTML editor a while ago. 😅

Reply:
Yeah I think Dave B and I are mildly concerned about this too. I'm going to do some testing.

Reply:
Maybe the observer could get replaced but not disconnected.  🤔